src/audio/dai-zephyr.c

```diff
@@ -811,58 +811,6 @@ static int dai_params(struct comp_dev *dev, struct sof_ipc_stream_params *params
 		dai_capture_params(dev, period_bytes, period_count);
 }
 
-static int dai_config_prepare(struct comp_dev *dev)
-{
-	struct dai_data *dd = comp_get_drvdata(dev);
-	int channel;
-
-	/* cannot configure DAI while active */
-	if (dev->state == COMP_STATE_ACTIVE) {
-		comp_info(dev, "dai_config_prepare(): Component is in active state.");
-		return 0;
-	}
-
-	if (!dd->dai_spec_config) {
-		comp_err(dev, "dai specific config is not set yet!");
-		return -EINVAL;
-	}
-
-	if (dd->chan) {
-		comp_info(dev, "dai_config_prepare(): dma channel index %d already configured",
-			  dd->chan->index);
-		return 0;
-	}
-
-	channel = dai_config_dma_channel(dev, dd->dai_spec_config);
-	comp_info(dev, "dai_config_prepare(), channel = %d", channel);
-
-	/* do nothing for asking for channel free, for compatibility. */
-	if (channel == DMA_CHAN_INVALID) {
-		comp_err(dev, "dai_config is not set yet!");
-		return -EINVAL;
-	}
-
-	/* get DMA channel */
-	channel = dma_request_channel(dd->dma->z_dev, &channel);
```

Comment:
looks like `dma_request_channel()` now isn't called at all, is that intentional? Would native Zephyr drivers builds still work after this?

Reply:
It is a mistake.  The wired thing is that if I use dma_request_channel with Zephyr driver, FW can't work.

Reply:
o, I got it. I used Zephyr built fw but not native Zephyr driver. Updated